src/repos.cc

```diff
@@ -999,7 +999,10 @@ void init_target( Zypper & zypper )
 
     try
     {
-      God->initializeTarget( zypper.globalOpts().root_dir );
+      if (zypper.globalOpts().install_root_dir.empty())
+        God->initializeTarget( zypper.globalOpts().root_dir );
+      else
+        God->initializeTarget( zypper.globalOpts().install_root_dir );
```

Comment:
I would not do it this way (i.e. just re-rooting the Target to the install_root_dir).

If you follow `--root` (usage of `root_dir`), you will find that several code locations refer to it when retrieving data (e.g. locks, product data,..). Re-rooting the Target alone is not sufficient for a correct workflow.

I'd start behaving exactly as if `--root` was passed (set root_dir, changedRoot,...). Just the `RepoManagerOptions` should be rooted to use `"/"` rather than `_gopts.root_dir`. 

This way you behave like `--root`, just the repos are taken from the outer system (this is also how we should describe it in the help and man page). Due to the repo refresh `--installroot` should require being `root`, which is not enforced with `--root`.

BTW both are mutual exclusive; there should be an error if --root and --installroot are specified at the same time..

Reply:
Done

Reply:
Done except the superuser part, do we need to enforce that? For example a zypper se inside the installroot would not require superuser privs?